pkg/synthesis/kubectl_test.go

```diff
@@ -0,0 +1,290 @@
+package synthesis
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"path"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	core "k8s.io/api/core/v1"
+	networking "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	admin "sigs.k8s.io/network-policy-api/apis/v1alpha1"
+
+	"github.com/np-guard/models/pkg/netset"
+
+	"github.com/np-guard/vmware-analyzer/internal/common"
+	analyzer "github.com/np-guard/vmware-analyzer/pkg/analyzer"
+	"github.com/np-guard/vmware-analyzer/pkg/collector"
+	"github.com/np-guard/vmware-analyzer/pkg/logging"
+)
+
+// This test is creating pods and policies and check the connections, using kubectl API.
+// The pods are created with a container that listen to TCP port 5000
+// The test is adjusted to this container:
+//   1. all TCP ports at the netpols are changed to tcp:5000
+//   2. connection between two pods is allowed iff the analyzed connection contains TCP
+
+func runK8STraceFlow(synTest *synthesisTest, t *testing.T, rc *collector.ResourcesContainerModel) {
+	if !hasKubectl() {
+		return
+	}
+	kubeDir := path.Join(synTest.debugDir(), "kube_test_dir")
+	err := logging.Tee(path.Join(kubeDir, "runK8STraceFlow.log"))
+	require.Nil(t, err)
+	k8sDir := path.Join(kubeDir, k8sResourcesDir)
+	setEvironmentFile := path.Join(kubeDir, "setEnvironment.sh")
+	cleanEvironmentFile := path.Join(kubeDir, "cleanEnvironment.sh")
+	// create K8S k8sResources
+	k8sResources, err := NSXToK8sSynthesis(rc, nil, synTest.options())
+	require.Nil(t, err)
+	// adjust k8sResources for the tests:
+	fixPodsResources(synTest.name, k8sResources.pods)
+	fixPoliciesResources(k8sResources.networkPolicies)
+	fixAdminPoliciesResources(k8sResources.adminNetworkPolicies)
+	require.Nil(t, k8sResources.CreateDir(kubeDir))
+	// run netpol-analizer, for debugging:
+	require.Nil(t, k8sAnalyzer(k8sDir, path.Join(kubeDir, "k8s_connectivity.txt"), "txt"))
+
+	// create the kubectl bash files:
+	require.Nil(t, createSetEvironmentFile(k8sDir, setEvironmentFile, k8sResources.pods))
+	require.Nil(t, createCleanEvironmentFile(cleanEvironmentFile))
+
+	// crreate environment:
+	logging.Debugf("creating environment from file %s", setEvironmentFile)
+	require.Nil(t, runCmdFile(setEvironmentFile))
+	logging.Debug("environment created")
+
+	// check connections:
+	checkErr := testConnections(synTest.name, kubeDir, rc)
+	// clean environmaent, we must clean before we checks for errors:
+	cleanErr := runCmdFile(cleanEvironmentFile)
```

Comment:
what does clean env includes?
will it delete all generated netpols? services? pods?

Reply:
services and policies are deleted, pods are are not deleted, to save time to the next test run

Reply:
then can you delete the pods after the last test run?

Reply:
added a flag - deletePodsAtCleanup with default true